### ISSUE TITLE
Vulkan back-end?

### ISSUE THREAD

**Issue by cesss** [PixarAnimationStudios/OpenSubdiv#1153]
I see there's no Vulkan back-end, but I believe it would be a great idea in many aspects:

- It's an API with concepts targeted to current gfx hardware designs, closer to Metal than to OpenGL.
- It's available on Windows, Linux, and MacOS (on MacOS you need the MoltenVK lib, which works on top of Metal, but I tried it and it works really great).
- Performance of OpenSubdiv should be better than with OpenGL.
- Gfx developers are progressively turning their eyes to Vulkan (just to put an example, even [ImGui](https://github.com/ocornut/imgui) has a Vulkan back-end).

I'm going to start Vulkan back-ends for my projects in the upcoming months, so I'd like that OpenSubdiv will be supported on them too.

**Comment by jilliene** [comment]
Filed as internal issue #OSD-298

**Comment by davidgyu** [comment]
Yes, we agree. We’ve been tracking progress on Vulkan (and MoltenVK) since the beginning and we are very excited about the opportunities there, but it hasn’t yet been a Studio priority. Much of the existing GLSL or HLSL shader can be used directly which simplifies the next steps of developing a Vulkan back-end implementation.

**Comment by jeroenbakker-atmind** [comment]
Hi, 

Here at Blender we are migrating towards Vulkan. The challenge is that we expect to use the OpenGL evaluator in Blender 3.1 (https://developer.blender.org/D12406) and when we introduce Vulkan we can't/won't want to use the OpenGL evaluator. Is there any news on this topic?

In our current plans the Vulkan migration will speed up on our side and are planning to have a Vulkan running in Blender within the next year. Perhaps we can see how to move forward with modern backends.

Regards,
Jeroen Bakker

**Comment by davidgyu** [comment]
Thank you for including that link to your work in progress. I haven't had a chance to look through your code yet, but the comments in that thread raise good questions and hopefully we can help you get things working well. We are definitely interested in having Blender take good advantage of OpenSubdiv.

We've been getting some good results using Vulkan and OpenSubdiv together including patch evaluation.

**Comment by bsavery** [comment]
Hi @davidgyu Any update on Vulkan support?

**Comment by RedMser** [comment]
Seems like this issue was supposed to be closed by #1320 but it was wrongly tagged?

v3.6.0 released with APIs to help supporting Vulkan: https://github.com/PixarAnimationStudios/OpenSubdiv/blob/v3_6_0/documentation/release_36.rst#modern-graphics-apis-and-parallel-computation

**Comment by NicTanghe** [comment]
AS a random end user it is unclear how for me how to build using vulcan.
